### ISSUE TITLE
[Bug] [Accessibility]: TextField component contrast issue with eye Icon in Dark, Aquatic and Desert theme

### ISSUE THREAD

**Issue by ebenshoshan** [microsoft/fluentui#35656]
### Package

react

### Package version

8.115.6

### React version

17.0.2

### Environment

```shell
Didn't work. Issue is also on fluent ui webpage:
https://developer.microsoft.com/en-us/fluentui#/controls/web/textfield:~:text=Password%20with%20reveal%20button
```

### Current Behavior

Contrast issues and visibility issues in Desert, Aquatic and Dark themes. Can't see the eye icon:

<img width="525" height="126" alt="Image" src="https://github.com/user-attachments/assets/49f68ad3-06dc-4d4f-ae79-37718d0e6313" />

<img width="537" height="133" alt="Image" src="https://github.com/user-attachments/assets/271622d0-b818-4ada-b360-062a29a7588f" />

<img width="529" height="126" alt="Image" src="https://github.com/user-attachments/assets/ace2c316-e170-460c-b290-43975ca7800a" />

Also the reveal/hide icon is not accessible through keyboard actions or voice command to enable/disable the show/hidden features through eye icon button.

### Expected Behavior

The icon should be visible and have good contrast in all supported themes.
The icon should be keyboard/voice command accessible.

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/textfield#feedback:~:text=Password%20with%20reveal%20button

### Steps to reproduce

1. Go to link.
2. Enable dark/aquatic/desert mode themes and see that button isn't visible in at least 1 of the 2 modes and that you can't transition to reveal button through keyboard or voice command.

### Are you reporting an Accessibility issue?

yes

### Suggested severity

Urgent - No workaround and Products/sites are affected

### Are you willing to submit a PR to fix?

yes

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a minimal reproducible example of the bug.

**Comment by ebenshoshan** [comment]
After further investigation, I believe this happens because of [this](https://github.com/microsoft/fluentui/blob/9745983cfa8656d24418bdf01386e9fc4243d774/packages/react/src/components/TextField/TextField.base.tsx#L681)

The textfield component enforces use of Edge's built in password reveal which doesn't hold up to accessibility standards.

**Comment by ebenshoshan** [comment]
Updated with explanation of other [bug](https://github.com/microsoft/fluentui/issues/35657)

**Comment by VasiliiOleinik** [comment]
Hi! I'd like to work on this issue if it's still open.
